haicrypt/hcrypt_xpt_srt.c

```diff
@@ -110,7 +109,22 @@ static void hcryptMsg_SRT_IndexMsg(unsigned char *msg, unsigned char *pfx_cache)
 	return; //nothing to do, header and index maintained by SRT
 }
 
-static int hcryptMsg_SRT_ParseMsg(unsigned char *msg)
+static const hcrypt_MsgInfo _hcMsg_SRT_MsgInfo = {
+	.hdr_len     = HCRYPT_MSG_SRT_HDR_SZ,
+	.pfx_len     = HCRYPT_MSG_SRT_PFX_SZ,
+	.getKeyFlags = hcryptMsg_SRT_GetKeyFlags,
+	.getPki      = hcryptMsg_SRT_GetPki,
+	.setPki      = hcryptMsg_SRT_SetPki,
+	.resetCache  = hcryptMsg_SRT_ResetCache,
+	.indexMsg    = hcryptMsg_SRT_IndexMsg
+};
+
+const hcrypt_MsgInfo* hcryptMsg_SRT_InitMsgInfo(void)
```

Comment:
But this "Init" function doesn't initialize anything. Just gets the pointer to the static global variable. Shouldn't be rather "Get" or something?

Reply:
Sorry, I rushed with the PR. It is just the same `hcryptMsg_SRT_MsgInfo` function that returns a pointer to the static variable.